token.go

```diff
@@ -178,3 +179,32 @@ func (m TokenModule) postTokenHandler(c *gin.Context) {
 
 	c.JSON(http.StatusCreated, token)
 }
+
+// postTokenHandler godoc
+// @summary Put token in database.
+// @description Put token in database. Provider in post object has to exists or should be empty.
+// @description Provider will has to be updated token ID during this operation.
+// @tags token
+// @accept json
+// @produce json
+// @param token body TokenWithProviderID _ "token object"
+// @success 201 {object} Token
+// @failure 400 {object} problem.Response "Bad request"
+// @failure 401 {object} problem.Response "Unauthorized or missing jwt token"
+// @failure 502 {object} problem.Response "Database is unreachable"
+// @router /token [put]
+func (m TokenModule) putTokenHandler(c *gin.Context) {
+	var token Token
+	if err := c.ShouldBindJSON(&token); err != nil {
+		problem.WriteInvalidBindError(c, err, "One or more parameters failed to parse when reading the request body.")
+		return
+	}
+	var placedToken Token
+	if err := m.Database.Where(token).FirstOrCreate(&placedToken).Error; err != nil {
+		problem.WriteInvalidBindError(c, err, fmt.Sprintf(
+			"Failed fetch or create on token with token:%q and username:%q.",
+			token.Token, token.UserName))
```

Comment:
Wrong problem template type, should be a DB write problem IMO.

And we shall never log nor report secrets.

```suggestion
		problem.WriteDBWriteError(c, err, fmt.Sprintf(
			"Failed fetch or create on token by username %q and token value.",
			token.UserName))
```

Reply:
Was blissfully unaware op the different template types. Perhaps we could report a token hash (say sha256)? It seems the usage of usernames here is unlikely so some way of identifying which token might be usefull.

Reply:
Hmm maybe, but then we get into the situation where should salt the hashes as well, but we don't have anything viable to salt it with. I'd say better off to just not include the token secret at all. This message is meant to be returned by the endpoint, and in the future we might want to log these problems as well.

We want to log and return as much information as possible just so that it's easier to understand the audit logs and error messages, but with the exception of secrets.

Reply:
```go
			"Failed fetch or create on token by username %q and token(sha256sum) %q.",
			sha256.Sum256([]byte(token.Token)),token.UserName))
```

Reply:
You're not salting the hash there. I consider this a security leak.

We don't gain much in even showing the token SHA hash other than some obscure correlation situations. I deem us better off if we don't include the token at all.

Reply:
Yeah with the salt i imagine it would be difficult to use it even just for correlations. Removed the hash code and mirrored your sugestion.